### PR TITLE
DiskBar: add constraint guide

### LINKED ISSUE #793: Clicking a partition in the custom install glitches the installer window
### What Happened?

Clicking a partition in the custom install glitches the installer window, which makes me being unable to set up partitions.

### Steps to Reproduce

1. Boot from Early Access 8 ISO (I used `elementaryos-8.0-daily.20240824.iso`)
2. Select the custom install
3. In the "Select Partitions" view, click "Modify Partitions…" to create partitions if none exists
4. In the "Select Partitions" view, click the storage bar (the yellow/green portion in the following screenshot)  
VirtualBox_test_25_08_2024_10_52_43
5. See the installer window get bigger and unable to setup partitions  
VirtualBox_test_25_08_2024_10_54_19

### Expected Behavior

The window should not glitch and the following popover should be shown where we can setup partitions:

VirtualBox_test_25_08_2024_10_57_47

### OS Version

8.x (Early Access)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
Aug 25 02:09:05 elementary io.elementary.g[1858]: Can't update stage views actor unnamed [MetaWindowActorX11] is on because it needs an allocation.
Aug 25 02:09:05 elementary io.elementary.g[1858]: Can't update stage views actor unnamed [MetaSurfaceActorX11] is on because it needs an allocation.
Aug 25 02:09:05 elementary io.elementary.i[2053]: gdk_monitor_get_geometry: assertion 'GDK_IS_MONITOR (monitor)' failed
Aug 25 02:09:05 elementary io.elementary.i[2053]: gdk_monitor_get_geometry: assertion 'GDK_IS_MONITOR (monitor)' failed
Aug 25 02:09:05 elementary io.elementary.i[2053]: gdk_monitor_get_geometry: assertion 'GDK_IS_MONITOR (monitor)' failed
Aug 25 02:09:05 elementary io.elementary.i[2053]: gtk_widget_size_allocate(): attempt to allocate GtkPopoverContent contents 0x562c36a38a20 with width 253 and height -23
Aug 25 02:09:05 elementary io.elementary.i[2053]: gtk_widget_size_allocate(): attempt to allocate GtkGrid grid 0x562c364c6b80 with width -2 and height -8
Aug 25 02:09:05 elementary io.elementary.installer.desktop[2053]: *** BUG ***
Aug 25 02:09:05 elementary io.elementary.installer.desktop[2053]: In pixman_region32_init_rect: Invalid rectangle passed
Aug 25 02:09:05 elementary io.elementary.installer.desktop[2053]: Set a breakpoint on '_pixman_log_error' to debug
```

[installer-glitch.log](https://github.com/user-attachments/files/16738721/installer-glitch.log)

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by danirabbit**
Fixes #793

### CODE REVIEW

**Review by ryonakano — COMMENTED** [review]
Installer crashes with `malloc_consolidate(): unaligned fastbin chunk detected` error when I re-entering the custom partition view:

https://github.com/user-attachments/assets/7cebcc65-6303-4148-95ce-aee5fb60ad2f

**Comment by danirabbit** [comment]
@ryonakano I think I finally did it! :pray:

**Review by ryonakano — COMMENTED** [review]
Sorry for my late review, confirmed the issue seems to be resolved in the live ISO too. Leaving one small comment

VirtualBox_test_12_10_2024_11_01_15

File: src/Views/PartitioningView.vala [hunk 1]
```diff
@@ -364,7 +364,7 @@ public class Installer.PartitioningView : AbstractInstallerView {
             device_path = "/dev/nvme0n1p2",
             filesystem = InstallerDaemon.FileSystem.LVM,
             start_sector = 542968,
-            end_sector = 976769070,
+            end_sector = 376769070,
```
[the remark this answers is not in the document]
Comment (danirabbit) [reply src/Views/PartitioningView.vala:367]: Ah no this was just to add another bar to test with more than two partitions. This is unrelated to the actual fix and the issue does occur with just this test changes in main still. I can revert this if you want

[the remark this answers is not in the document]
Comment (ryonakano) [reply src/Views/PartitioningView.vala:367]: I'd like you to drop this change then :bow:

[the remark this answers is not in the document]
Comment (danirabbit) [reply src/Views/PartitioningView.vala]: Done!

**Review by ryonakano — APPROVED** [review]
Thank you very much for this fix!

**Comment by danirabbit** [comment]
@ryonakano thanks for reviewing all the different branches I made trying to figure it out 😅

**Pull request merged**